lms/djangoapps/lms_xblock/runtime.py

```diff
@@ -66,64 +66,68 @@ def unquote_slashes(text):
-
-    This must be mixed in to a runtime that already accepts and stores
-    a course_id
-    """
-    # pylint: disable=unused-argument
-    def handler_url(self, block, handler_name, suffix='', query='', thirdparty=False):
-        """See :method:`xblock.runtime:Runtime.handler_url`"""
-        view_name = 'xblock_handler'
-        if handler_name:
-            # Be sure this is really a handler.
-            func = getattr(block, handler_name, None)
-            if not func:
-                raise ValueError("{!r} is not a function name".format(handler_name))
-            if not getattr(func, "_is_xblock_handler", False):
-                raise ValueError("{!r} is not a handler name".format(handler_name))
-
-        if thirdparty:
-            view_name = 'xblock_handler_noauth'
-
-        url = reverse(view_name, kwargs={
-            'course_id': unicode(self.course_id),
-            'usage_id': quote_slashes(unicode(block.scope_ids.usage_id).encode('utf-8')),
-            'handler': handler_name,
-            'suffix': suffix,
-        })
-
-        # If suffix is an empty string, remove the trailing '/'
-        if not suffix:
-            url = url.rstrip('/')
-
-        # If there is a query string, append it
-        if query:
-            url += '?' + query
-
-        # If third-party, return fully-qualified url
-        if thirdparty:
-            scheme = "https" if settings.HTTPS == "on" else "http"
-            url = '{scheme}://{host}{path}'.format(
-                scheme=scheme,
-                host=settings.SITE_NAME,
-                path=url
-            )
-
-        return url
-
-    def local_resource_url(self, block, uri):
-        """
-        local_resource_url for Studio
-        """
-        path = reverse('xblock_resource_url', kwargs={
-            'block_type': block.scope_ids.block_type,
-            'uri': uri,
-        })
-        return '//{}{}'.format(settings.SITE_NAME, path)
+def handler_url(block, handler_name, suffix='', query='', thirdparty=False):
+    """
+    This method matches the signature for `xblock.runtime:Runtime.handler_url()`
+
+    See :method:`xblock.runtime:Runtime.handler_url`
+    """
+    view_name = 'xblock_handler'
+    if handler_name:
+        # Be sure this is really a handler.
+        #
+        # We're checking the .__class__ instead of the block itself to avoid
+        # auto-proxying from Descriptor -> Module, in case descriptors want
+        # to ask for handler URLs without a student context.
+        func = getattr(block.__class__, handler_name, None)
+        if not func:
+            raise ValueError("{!r} is not a function name".format(handler_name))
+
+        # Is the following necessary? ProxyAttribute causes an UndefinedContext error
+        # if trying this without the module system.
+        #
+        #if not getattr(func, "_is_xblock_handler", False):
+        #    raise ValueError("{!r} is not a handler name".format(handler_name))
```

Comment:
I remember fiddling with this for a bit and not finding a simple way to check to see if that function had that attribute via its' `__class__` (and thus avoiding the proxy magic). There might be another way to get at it, but I unless it's really necessary, I'd prefer to do nothing at all rather than to add yet more magic to work around existing magic. :-(